plugin/src/main/java/io/jenkins/plugins/casc/ConfigurationAsCode.java

```diff
@@ -581,7 +581,7 @@ private void configureWith(List<YamlSource> sources) throws ConfiguratorExceptio
         final PathMatcher matcher = FileSystems.getDefault().getPathMatcher(YAML_FILES_PATTERN);
         try (Stream<Path> stream = Files.find(Paths.get(path), Integer.MAX_VALUE,
                 (next, attrs) -> !attrs.isDirectory() && !isHidden(next) && matcher.matches(next))) {
-            return stream.collect(toList());
+            return stream.sorted().collect(toList());
```

Comment:
if you add a test there's less chance of this breaking at some point in the future

Reply:
I've added a simple positive test that I believe tests this.